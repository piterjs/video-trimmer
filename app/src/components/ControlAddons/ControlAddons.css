.Control-addons {
    max-width: 100%;
    position: relative;
    display: inline-flex;
    vertical-align: middle;
}
.Control-addons > .Control {
    position: relative;
    flex: 1 1 auto;
}
.Control-addons > .Control:hover,
.Control-addons > .Control:focus,
.Control-addons > .Control:active {
    z-index: 1;
}
.Control-addons > .Control:first-child {
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
}
.Control-addons > .Control:not(:first-child) {
    margin-left: -1px;
}
.Control-addons > .Control:not(:first-child):not(:last-child) {
    border-radius: 0;
}
.Control-addons > .Control:last-child {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
}
