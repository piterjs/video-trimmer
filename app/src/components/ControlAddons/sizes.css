/* sizes */
.Control-addons--size-small > .Control {
    font-size: .75rem;
    border-radius: 2px;
}
.Control-addons--size-normal > .Control {
    font-size: 1rem;
    border-radius: 4px;
}
.Control-addons--size-medium > .Control {
    font-size: 1.25rem;
    border-radius: 4px;
}
.Control-addons--size-large > .Control {
    font-size: 1.5rem;
    border-radius: 4px;
}

.Control-addons.is-fullwidth > .Control {
    width: 100%;
}
