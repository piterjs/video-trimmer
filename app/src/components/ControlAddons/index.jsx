import React from 'react';
import './ControlAddons.css';
import './sizes.css';

function ControlAddons ({size = 'normal', color = 'default', tag = 'button', className = '', ...rest}) {
    const classes = [
        'Control-addons',
        `Control-addons--size-${size}`,
        className
    ].join(' ').trim()

    return <div
        role="group"
        className={classes}
        {...rest}
    >{rest.children}</div>
}

export default ControlAddons;
