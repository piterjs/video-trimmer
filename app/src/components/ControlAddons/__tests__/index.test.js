import React from 'react';
import renderer from 'react-test-renderer';
import ControlAddons from '../';

describe('ControlAddons', () => {
    it('should be defined', () => {
        expect(ControlAddons).toBeDefined();
    });
    it('should render correctly', () => {
        const tree = renderer
        .create(<ControlAddons />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
