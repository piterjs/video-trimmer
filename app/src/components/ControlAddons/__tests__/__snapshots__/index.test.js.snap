// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlAddons should render correctly 1`] = `
<div
  className="Control-addons Control-addons--size-normal"
  role="group"
/>
`;
